kong/tracing/instrumentation.lua

```diff
@@ -72,33 +72,32 @@ function _M.balancer(ctx)
     local try = balancer_tries[i]
     span = tracer.start_span("balancer try #" .. i, {
       kind = 3, -- client
-      start_time_ns = try.balancer_start * 100000,
+      start_time_ns = try.balancer_start * 1e6,
       attributes = {
-        ["kong.balancer.state"] = try.state,
-        ["http.status_code"] = try.code,
         ["net.peer.ip"] = try.ip,
         ["net.peer.port"] = try.port,
       }
     })
 
-    if i < try_count then
+    if try.state then
+      span:set_attribute("http.status_code", try.code)
       span:set_status(2)
     end
 
     -- last try
-    if i == try_count then
-      local upstream_finish_time = (ctx.KONG_BODY_FILTER_ENDED_AT or 0) * 100000
+    if i == try_count and try.state == nil then
+      local upstream_finish_time = ctx.KONG_BODY_FILTER_ENDED_AT and ctx.KONG_BODY_FILTER_ENDED_AT * 1e6
       span:finish(upstream_finish_time)
 
     else
       if not upstream_connect_time then
-        upstream_connect_time = split(var.upstream_connect_time, ",", "jo")
+        upstream_connect_time = split(var.upstream_connect_time, ", ", "jo")
       end
 
       -- retrying
       if try.balancer_latency ~= nil then
-        local try_upstream_connect_time = (upstream_connect_time[i] or 0) * 1000
-        span:finish((try.balancer_start + try.balancer_latency + try_upstream_connect_time) * 100000)
+        local try_upstream_connect_time = (tonumber(upstream_connect_time[i]) or 0) * 1000
```

Comment:
```suggestion
        local try_upstream_connect_time = (tonumber(upstream_connect_time[i] or 0)) * 1000
```

to catch the case where its value is `"non-number-string"`

Reply:
Actually, the original case was to avoid the "non-number-string".
`tonumber` on a non-number string will result in a nil value.

```bash
$ resty -e "print(type(tonumber('-')))"
nil
```